File: with_elixir/test/day4_test.exs
defmodule Day4Test do
  use ExUnit.Case
  doctest Day4

  import ReadInput

  test "part1..." do
    assert Day4.part1(read_input("day4sample")) == 18
  end

  test "part2..." do
    assert Day4.part2(read_input("day4sample")) == 9
  end
end

File: 2024/with_node/src/day4.test.ts
import assert from "node:assert";
import test from "node:test";

import { readInput } from "./readInput.ts";
import { part1, part2 } from "./day4.ts";

test("day4...", async (t) => {
  await t.test("part1 says...", () => {
    assert.strictEqual(part1(readInput("day4sample")), 18);
  });

  await t.test("part2 says...", () => {
    assert.strictEqual(part2(readInput("day4sample")), 9);
  });
});
